sections/sprint-4/updated-uppaal.tex

```diff
@@ -164,32 +164,39 @@ \subsection*{UDP host templates}
 \end{figure}
 
 % Trace example of the system
-\subsection*{Perfect scenario simulation trace}
-In \uppaal it is possible to generate traces of simulations of the model. 
-This is useful when verifying different properties of the model or when getting an overview of how the model works.
-A simulation trace can be seen on \autoref{fig:sprint4-happy-trace}.
-At the top of the trace you see all of the processes in the system.
-You can then see the trace of each process which is the line that goes from the process to the bottom of the diagram.
-The boxes on the trace indicate which \uppLoc{Location} the process is in at any given time.
-The red arrows between the traces symbolize synchronizations that happen between processes, and the label on the red arrow indicates which channel it happens through.
-As seen on the trace \uppTemp{Client(0)} first connects through the threeway handshake without any timeouts.
-\uppTemp{Client(1)} then also connects through a three-way handshake without any timeouts.
-Finally, after both of the clients are in the \uppLoc{Lobby}, the \uppTemp{Host} transitions to \uppLoc{InGame}.
+\subsection{Verification of the models}
+In \uppaal it is also possible to perform verifications on the model to ensure that some properties hold true for the model.
+This is done by creating a query that the \uppaal engine can then verify on the model.
+It is then possible to get a trace of that query that either proves or disproves the query.
+To demonstrate this a query was verified on the application-level model and the full model.
+The property that was verified was \uppEF{Host.InGame and forall(i:id_t) Client(i).InGame}.
+
+\subsubsection{Verification of the application-level model}
+The property held true for the model an the trace that was produced can be seen on \autoref{fig:sprint4-verification-application-level}.
+As can be seen on the trace, all of the clients go through the process of connecting without timing out.
+When all of the clients have connected, the \uppProc{Host} starts the game through the \uppSync{TCP_start_game} synchronization and the host and all of the clients transition to the \uppLoc{InGame} location.
 \begin{figure}[H]
     \centering
-    \includegraphics[width=1\linewidth]{sprint4/connected-trace.png}
-    \caption{Example of a trace where both \uppProc{Client} processes connect without any problems.}
-    \label{fig:sprint4-happy-trace}
+    \includegraphics[width=0.8\linewidth]{sprint4/application-level-verification.png}
+    \caption{Verification trace of the application-level model.}
```

Comment:
måske skrive at tracet viser hvordan man kan gå fra start og frem til hvor host er ingame og alle andre er ingame samtidig? Måske at det er the shortest trace?